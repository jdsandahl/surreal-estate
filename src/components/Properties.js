import React, { useState, useEffect } from "react";
import PropertyListings from "./PropertyListings";
import axios from "axios";

const Properties = () => {
  const initialState = {
    listings: [],
    loading: false,
  };

  const [listings, setListings] = useState(initialState.listings);
  const [loading, setLoading] = useState(initialState.loading);

  useEffect(() => {
    setLoading(true);

    axios
      .get("http://localhost:4000/api/v1/PropertyListing")
      .then((response) => {
        setListings(response.data);
        setLoading(false);
      });
  }, []);

  if (loading) {
    return <div>loading</div>;
  }
  return (
    <div>
      <PropertyListings listings={listings} />
    </div>
  );
};

export default Properties;
