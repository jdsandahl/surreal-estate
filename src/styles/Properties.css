.properties {
  background-color: rgb(230, 230, 230);
  display: grid;
  grid-template-columns: auto 1fr;
  grid-template-rows: auto;
}
