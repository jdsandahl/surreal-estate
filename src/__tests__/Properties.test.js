import React from "react";
import { render, waitForElementToBeRemoved, act } from "@testing-library/react";
import Properties from "../components/Properties";

describe("Properties", () => {
  test("the page renders correctly against snapshot", () => {
    const { asFragment, getByText, getByRole } = render(<Properties />);
    const component = asFragment();

    expect(component).toMatchSnapshot();
  });

});
