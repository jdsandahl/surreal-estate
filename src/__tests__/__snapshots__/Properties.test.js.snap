// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties the page renders correctly against snapshot 1`] = `
<DocumentFragment>
  <div
    class="properties"
  >
    <div
      class="properties__main"
    >
      <div
        class="loading"
      >
        <div
          class="indeterminate"
        />
      </div>
      <div
        class="property-listings"
      />
    </div>
    <div
      class="sidebar"
    >
      <div
        class="burger"
      >
        <input
          class="burger__toggle"
          type="checkbox"
        />
        <div
          class="burger__lines"
        >
          <div />
        </div>
      </div>
      <div
        class="sidebar__filters--collapsed"
      >
        <form
          class="search-form"
        >
          <input
            class="search-form__input"
            type="text"
            value=""
          />
          <button
            class="search-form__submit"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-search fa-w-16 "
              data-icon="search"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
              />
            </svg>
          </button>
        </form>
        <label
          class="sidebar__section"
        >
          Filter By City:
          <ul
            class="sidebar__links"
          >
            <li
              class="sidebar__item"
            >
              <a
                class="sidebar__link"
                href="/properties"
              >
                All
              </a>
            </li>
            <li
              class="sidebar__item"
            >
              <a
                class="sidebar__link"
                href="/?query={\\"city\\":\\"Manchester\\"}"
              >
                Manchester
              </a>
            </li>
            <li
              class="sidebar__item"
            >
              <a
                class="sidebar__link"
                href="/?query={\\"city\\":\\"Leeds\\"}"
              >
                Leeds
              </a>
            </li>
            <li
              class="sidebar__item"
            >
              <a
                class="sidebar__link"
                href="/?query={\\"city\\":\\"Sheffield\\"}"
              >
                Sheffield
              </a>
            </li>
            <li
              class="sidebar__item"
            >
              <a
                class="sidebar__link"
                href="/?query={\\"city\\":\\"Liverpool\\"}"
              >
                Liverpool
              </a>
            </li>
          </ul>
        </label>
        <label
          class="sidebar__section"
        >
          Sort By:
          <ul
            class="sidebar__links"
          >
            <li
              class="sidebar__item"
            >
              <a
                class="sidebar__link"
                href="/?sort={\\"price\\":1}"
              >
                Price - from lowest
              </a>
            </li>
            <li
              class="sidebar__item"
            >
              <a
                class="sidebar__link"
                href="/?sort={\\"price\\":-1}"
              >
                Price - from highest
              </a>
            </li>
          </ul>
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
